### PR TITLE
test_host_resolve: rewrite test cases in criterion

### LINKED ISSUE #2610: convert test_host_resolve to criterion
Use Criterion to rewrite this unit test.

### PR DESCRIPTION
**Pull request by mehul-m-prajapati**
fixes #2610

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by MrAnno** [comment]
@kira-syslogng ok to test

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
From 50539cf79002ab0c94c07f46996507517bcedf3b:

> This test program is not run automatically in make check invocations as it
relies on DNS and that's not always available where syslog-ng is being compiled.

I think, in this case, `noinst_PROGRAMS` should be left untouched.

**Comment by mehul-m-prajapati** [comment]
Hi @MrAnno,

Thanks for a quick review.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/tests/test_host_resolve.c [hunk 1]
```diff
@@ -296,22 +248,21 @@ test_hostname_is_normalized_if_normalize_hostnames_is_set(void)
 }
 
 static void
-test_resolve_hostname_to_hostname(void)
+setup(void)
 {
-  HOST_RESOLVE_TESTCASE(test_short_hostname_is_converted_to_fqdn_if_use_fqdn_is_set);
-  HOST_RESOLVE_TESTCASE(test_fqdn_hostname_is_converted_to_short_if_use_fqdn_is_unset);
-  HOST_RESOLVE_TESTCASE(test_hostname_is_normalized_if_normalize_hostnames_is_set);
+  app_startup();
```
Comment (furiel) [inline lib/tests/test_host_resolve.c]: Do we need `app_startup`?

Comment (mehul-m-prajapati) [reply lib/tests/test_host_resolve.c]: Yes, I think it is not needed.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
The previous failure was a CI issue, nothing related to the changeset of this PR.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by furiel** [comment]
@mehul-m-prajapati thank you!

**Comment by MrAnno** [comment]
@mehul-m-prajapati Thanks.

**Pull request merged**